pkg/operator/controllers/machine/machine_controller_test.go

```diff
@@ -24,167 +24,47 @@ import (
-}`)},
-				},
-			},
-		}
+	// Fake cluster missing a master
+	newFakeMao2 := func(diskSize, imagePublisher, vmSize, masterDiskSize, masterVmSize string) *maofake.Clientset {
+		master0 := masterMachine("foo-hx8z7-master-0", "", "", "", "")
+		master2 := masterMachine("foo-hx8z7-master-2", "", "", "", "")
+		worker0 := workerMachine("foo-hx8z7-worker-0", "", "", "", "")
+		worker1 := workerMachine("foo-hx8z7-worker-1", "", "", "", "")
+		worker2 := workerMachine("foo-hx8z7-worker-2", "", "", "", "")
+		workerMachineSet0 := workerMachineSet("foo-hx8z7-machineset-0", 1)
+		workerMachineSet1 := workerMachineSet("foo-hx8z7-machineset-1", 1)
+		workerMachineSet2 := workerMachineSet("foo-hx8z7-machineset-2", 1)
 
-		worker0 := &machinev1beta1.Machine{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "foo-hx8z7-worker-0",
-				Namespace: machineSetsNamespace,
-				Labels:    map[string]string{"machine.openshift.io/cluster-api-machine-role": "worker"},
-			},
-			Spec: machinev1beta1.MachineSpec{
-				ProviderSpec: machinev1beta1.ProviderSpec{
-					Value: &runtime.RawExtension{
-						Raw: []byte(fmt.Sprintf(`{
-"apiVersion": "azureproviderconfig.openshift.io/v1beta1",
-"kind": "AzureMachineProviderSpec",
-"osDisk": {
-"diskSizeGB": %v
-},
-"image": {
-"publisher": "%v",
-"offer": "aro4"
-},
-"vmSize": "%v"
-}`, diskSize, imagePublisher, vmSize))},
-				},
-			},
-		}
-
-		worker1 := &machinev1beta1.Machine{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "foo-hx8z7-worker-1",
-				Namespace: machineSetsNamespace,
-				Labels:    map[string]string{"machine.openshift.io/cluster-api-machine-role": "worker"},
-			},
-			Spec: machinev1beta1.MachineSpec{
-				ProviderSpec: machinev1beta1.ProviderSpec{
-					Value: &runtime.RawExtension{
-						Raw: []byte(`{
-"apiVersion": "azureproviderconfig.openshift.io/v1beta1",
-"kind": "AzureMachineProviderSpec",
-"osDisk": {
-"diskSizeGB": 128
-},
-"image": {
-"publisher": "azureopenshift",
-"offer": "aro4"
-},
-"vmSize": "Standard_D4s_v3"
-}`)},
-				},
-			},
-		}
+		return maofake.NewSimpleClientset(worker0, worker1, worker2, master0, master2, workerMachineSet0, workerMachineSet1, workerMachineSet2)
+	}
 
-		worker2 := &machinev1beta1.Machine{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "foo-hx8z7-worker-2",
-				Namespace: machineSetsNamespace,
-				Labels:    map[string]string{"machine.openshift.io/cluster-api-machine-role": "worker"},
-			},
-			Spec: machinev1beta1.MachineSpec{
-				ProviderSpec: machinev1beta1.ProviderSpec{
-					Value: &runtime.RawExtension{
-						Raw: []byte(`{
-"apiVersion": "azureproviderconfig.openshift.io/v1beta1",
-"kind": "AzureMachineProviderSpec",
-"osDisk": {
-"diskSizeGB": 128
-},
-"image": {
-"publisher": "azureopenshift",
-"offer": "aro4"
-},
-"vmSize": "Standard_D4s_v3"
-}`)},
-				},
-			},
-		}
+	// Fake cluster missing a worker
+	newFakeMao3 := func(diskSize, imagePublisher, vmSize, masterDiskSize, masterVmSize string) *maofake.Clientset {
```

Comment:
since this is only used to test the number of workers, you can omit these unused params
```suggestion
	newFakeMao3 := func() *maofake.Clientset {
```